app.py

```diff
@@ -66,44 +66,88 @@ def adduser():
     return jsonify({'id' : new_user.user_id}), 201
 
 
-@app.route('/homecookedfood/addtiffin', methods=['POST'])
+@app.route('/hcf/addmeal', methods=['POST'])
 @auth.login_required
-def addtiffin():
+def addmeal():
     content = request.json
     user_id = g.user.user_id
 
-    if (not content or not 'tiffin_details' in content):
+    if (not content or not 'meal_details' in content or not price_per_meal in content):
         abort(400)
 
-    tiffin = models.Tiffins(user_id, content['tiffin_details'], max_tiffins = content.get('max_tiffins', 0))
+    meal = models.Meal(user_id, content['meal_details'], content['price_per_meal'], max_meals = content.get('max_meals', 0))
 
-    db.session.add(tiffin)
+    g.user.make_provider() 
+
+    db.session.add(meal)
     db.session.commit()
     return jsonify({}), 201
 
+
+@app.route('/hcf/getprovidersbyzipcode', methods=['GET'])
+@auth.login_required
+def getprovidersbyzipcode():
+    content = request.json
     
-@app.route('/homecookedfood/gettiffinsbyzipcode', methods=['GET'])
+    if (not content or not 'zipcode' in content):
+        abort(400)
+
+    users = models.Users.query.filter_by(zipcode=content['zipcode'], is_provider=True)
+    return jsonify({'list_of_providers': users}), 201
+
+
+@app.route('/hcf/getmealsbyprovider', methods=['GET'])
 @auth.login_required
-def gettiffinsbyzipcode():
+def getmealsbyprovider():
+    content = request.json
+    
+    if (not content or not 'provider_id' in content):
+        abort(400)
+
+    user = models.Users.query.filter_by(user_id=content['provider_id']).first()
+    return jsonify({'meals_by_provider': user.meals}), 201
+
+    
+@app.route('/hcf/getmealsbyzipcode', methods=['GET'])
+@auth.login_required
+def getmealsbyzipcode():
     content = request.json
     
     if (not content or not 'zipcode' in content):
         abort(400)
 
-    tiffins = []
+    meals = []
     users = models.Users.query.filter_by(zipcode=content['zipcode'])
     for user in users:
-        list_of_tiffins_user = user.tiffins
-        for tiffin in list_of_tiffins_user:
-            tiffins.append((tiffin.tiffin_id, tiffin.tiffin_details, tiffin.max_tiffins))        
+        list_of_meals_user = user.meals
+        for meal in list_of_meals_user:
+            meals.append(meal)
+
+    return jsonify({'list_of_meals': meals}), 201
+
+
+@app.route('/hcf/givecommenttoprovider', methods=['POST'])
+@auth.login_required
+def givecommenttoprovider():
+    content = request.json
+    
+    if (not content or not 'provider_id' in content):
```

Comment:
I can see if there is a better way to make sure the request is correct, but if not, this is repetitive but its different for each request, so not sure if it can be easily put into a method.